providers/flagd/pkg/service/service.go

```diff
@@ -0,0 +1,180 @@
+	}))
+	if err != nil {
+		return &schemaV1.ResolveStringResponse{
+			Reason: string(of.ErrorReason),
+		}, errors.New(handleError(err))
+	}
+	return res.Msg, nil
+}
+
+// ResolveFloat handles the flag evaluation response from the  flagd interface ResolveFloat rpc
+func (s *Service) ResolveFloat(ctx context.Context, flagKey string, evalCtx map[string]interface{}) (*schemaV1.ResolveFloatResponse, error) {
+	client := s.Client.Instance()
+	if client == nil {
+		return &schemaV1.ResolveFloatResponse{
+			Reason: flagdModels.ErrorReason,
+		}, of.NewProviderNotReadyResolutionError(ConnectionError)
+	}
+	evalCtxF, err := structpb.NewStruct(evalCtx)
+	if err != nil {
+		log.Error(err)
+		return &schemaV1.ResolveFloatResponse{
+			Reason: flagdModels.ErrorReason,
+		}, of.NewParseErrorResolutionError(err.Error())
+	}
+	res, err := client.ResolveFloat(ctx, connect.NewRequest(&schemaV1.ResolveFloatRequest{
+		FlagKey: flagKey,
+		Context: evalCtxF,
+	}))
+	if err != nil {
+		return &schemaV1.ResolveFloatResponse{
+			Reason: string(of.ErrorReason),
+		}, errors.New(handleError(err))
+	}
+	return res.Msg, nil
+}
+
+// ResolveInt handles the flag evaluation response from the  flagd interface ResolveNumber rpc
+func (s *Service) ResolveInt(ctx context.Context, flagKey string, evalCtx map[string]interface{}) (*schemaV1.ResolveIntResponse, error) {
+	client := s.Client.Instance()
+	if client == nil {
+		return &schemaV1.ResolveIntResponse{
+			Reason: flagdModels.ErrorReason,
+		}, of.NewProviderNotReadyResolutionError(ConnectionError)
+	}
+	evalCtxF, err := structpb.NewStruct(evalCtx)
+	if err != nil {
+		log.Error(err)
+		return &schemaV1.ResolveIntResponse{
+			Reason: flagdModels.ErrorReason,
+		}, of.NewParseErrorResolutionError(err.Error())
+	}
+	res, err := client.ResolveInt(ctx, connect.NewRequest(&schemaV1.ResolveIntRequest{
+		FlagKey: flagKey,
+		Context: evalCtxF,
+	}))
+	if err != nil {
+		return &schemaV1.ResolveIntResponse{
+			Reason: string(of.ErrorReason),
+		}, errors.New(handleError(err))
+	}
+	return res.Msg, nil
+}
+
+// ResolveObject handles the flag evaluation response from the  flagd interface ResolveObject rpc
+func (s *Service) ResolveObject(ctx context.Context, flagKey string, evalCtx map[string]interface{}) (*schemaV1.ResolveObjectResponse, error) {
+	client := s.Client.Instance()
+	if client == nil {
+		return &schemaV1.ResolveObjectResponse{
+			Reason: flagdModels.ErrorReason,
+		}, of.NewProviderNotReadyResolutionError(ConnectionError)
+	}
+	evalCtxF, err := structpb.NewStruct(evalCtx)
+	if err != nil {
+		log.Error(err)
+		return &schemaV1.ResolveObjectResponse{
+			Reason: flagdModels.ErrorReason,
+		}, of.NewParseErrorResolutionError(err.Error())
+	}
+	res, err := client.ResolveObject(ctx, connect.NewRequest(&schemaV1.ResolveObjectRequest{
+		FlagKey: flagKey,
+		Context: evalCtxF,
+	}))
+	if err != nil {
+		return &schemaV1.ResolveObjectResponse{
+			Reason: string(of.ErrorReason),
+		}, errors.New(handleError(err))
```

Comment:
is it possible to return these errors as an openfeature.ResolutionError so they can be picked up on at the outer level?

Reply:
done :)